qiskit/synthesis/clifford/clifford_decompose_layers.py

```diff
@@ -203,24 +204,25 @@ def _create_graph_state(cliff, validate=False):
     """
 
     num_qubits = cliff.num_qubits
-    rank = _compute_rank(cliff.stab_x)
+    rank = compute_rank((cliff.stab_x).astype(bool))
```

Comment:
You should do this to avoid a copy since the stabilizer slice should be of `dtype=bool` already.
```suggestion
    rank = compute_rank((cliff.stab_x).astype(bool, copy=False))
```

That being said I typically prefer to do:

```suggestion
    rank = compute_rank(np.asarray(cliff.stab_x, dtype=bool))
```

because it defaults to no copy behavior if all the data is aligned.